### DISCUSSION TITLE
Unexpected crash when octree is called

### DISCUSSION THREAD

**Discussion by qiqigit** [libigl/libigl#2038] [Q&A]
I came across unexpected crash when I tried to build an octree ( before calling the igl::knn function).  The code snippet is as below.
I would appreciate it if anyone could provide any help to solve this. Thanks a lot!
-------------------------------------------------------
    Eigen::MatrixXd V;  Eigen::MatrixXi F;
    igl::readOBJ(fname, V, F);  // A triangular mesh with 20K polygons
    std::vector<std::vector<int > > O_PI;
    Eigen::MatrixXi O_CH;
    Eigen::MatrixXd O_CN;
    Eigen::VectorXd O_W;
    igl::octree(V, O_PI, O_CH, O_CN, O_W);

**Comment by mathsyouth** [comment]
I ran into the same issue.

**Comment by alecjacobson** [comment]
Could you please share a problematic input mesh?

**Reply by mathsyouth** [reply]
The problematic mesh is [input_mesh.zip](https://github.com/libigl/libigl/files/11929482/input_mesh.zip).

**Reply by mathsyouth** [reply]
```
Eigen::MatrixXd V;
Eigen::MatrixXi F;
const std::string inputMesh =  "upper_step1._1T.stl";
igl::read_triangle_mesh(inputMesh, V, F);
std::vector<std::vector<int>> O_PI;
Eigen::MatrixXi O_CH;
Eigen::MatrixXd O_CN;
Eigen::VectorXd O_W;
igl::octree(V, O_PI, O_CH, O_CN, O_W);
```
